### PR TITLE
Pixel Side Scroller: Tilemap

### PR DESCRIPTION
**Pull request by Numenter**
# Description
Should "Pixel Side Scroller" use Tilemaps?
image
Adds:
- added rock tileset
- only first map changed at this point

Bugs:
- flicking tileset (probably caused by smooth camera transition)
[fixed with use_gpu_pixel_snap, could impact other games, can't change on runtime]

## Type of change
- [x] New feature (non breaking change that adds functionality)

### CODE REVIEW

**Comment by Joshix-1** [comment]
I really like the current design with the text barriers.

I think the version with tilemaps should be another game.

**Comment by RedstoneMedia** [comment]
I think so too. This seems too different

**Comment by MarioLiebisch** [comment]
I agree with the others, I'm feeling mixed. Using a tilemap certainly isn't wrong here, but should it use a more generic tileset or keep the look?

**Comment by Numenter** [comment]
The text barriers are unique but also boring to look at (my opinion). Maybe something like this?
image

**Comment by MarioLiebisch** [comment]
Okay, have to say, this looks way more interesting now :)

**Comment by ASecondGuy** [comment]
I think something like in dwarf fortress could fit very well.

**Comment by Numenter** [comment]
Pushed this up to a PR. There are some missing parts, like the traps and the other maps.
I pledge to update those in a followup PR.

- Text barriers style is limiting.
- Node/Label based level creation is fiddly and slow in prototyping.
- Branching Pixel Side Scroller is splitting interests and I can guarantee one of them will be a dust catcher, just like all the other games.

File: game/games/pixel_side_scroller/pawns/character/character.tscn [hunk 1]
```diff
@@ -75,7 +75,7 @@ tracks/0/keys = {
 }
 
 [sub_resource type="RectangleShape2D" id=4]
-extents = Vector2( 4, 7.5 )
+extents = Vector2( 3.94, 7.5 )
```
Comment (MarioLiebisch) [inline game/games/pixel_side_scroller/pawns/character/character.tscn:78]: I assume this change was done so the character properly fits through narrow 1 tile wide gaps?

Comment (Numenter) [reply game/games/pixel_side_scroller/pawns/character/character.tscn:78]: yes

**Pull request merged**